builtins/amp-img.js

```diff
@@ -167,6 +177,31 @@ export class AmpImg extends BaseElement {
     return true;
   }
 
+  /**
+   * Checks to see if there is a placeholder that is blurred
+   * @private
+   */
+  checkBlur_() {
+    const placeholder = this.getPlaceholder();
+    if (!!placeholder && placeholder.classList.contains('i-amphtml-blur')) {
+      this.hasBlurredPlaceHolder_ = true;
+      this.element.classList.add('i-amphtml-blur-loading');
+    }
+  }
+
+  /**
+   * @private
+   * Animates the fully rendered either through a fade in on top of a blurry
+   * placeholder if one exists, or hiding the fallback if not
+   */
+  renderImg_() {
+    if (this.hasBlurredPlaceHolder_) {
```

Comment:
You won't need this ivar if you just check `this.classList.contains('i-amphtml-blur-loading')`.